src/ProcessManager.cpp

```diff
@@ -0,0 +1,45 @@
+#include "ProcessManager.h"
+#include <iostream>
+
+void ProcessManager::addBackgroundProcess(pid_t pid) {
+    backgroundProcesses.push_back(pid);
+}
+
+void ProcessManager::showJobs() {
+    std::cout << "Running background processes:" << std::endl;
+    for (auto it = backgroundProcesses.begin(); it != backgroundProcesses.end();) {
+        pid_t pid = *it;
+        int childStatus;
+        pid_t result = waitpid(pid, &childStatus, WNOHANG | WUNTRACED | WCONTINUED);
+
+        if (result == -1) {
+            std::cout << "PID: " << pid << ", Status: Not Found!" << std::endl;
+            it = backgroundProcesses.erase(it);
+        }
+        else if (result == 0) {
+            // Process still running
+            std::cout << "PID: " << pid << ", Status: Running..." << std::endl;
+            ++it;
+        }
+        else {
+            // Process finished
+            if (WIFEXITED(childStatus)) {
+                std::cout << "PID: " << pid << ", Status: Exited." << std::endl;
```

Comment:
You were asked to also show the command of this process. Think about which info is helpful for a user who want to see the background jobs. Is just PID enough?